src/ItemGroup.sip

```diff
@@ -0,0 +1,77 @@
+//Copyright (c) 2020 Ultimaker B.V.
+//pynest2d is released under the terms of the LGPLv3 or higher.
+
+%MappedType ItemGroup
+{
+    %TypeHeaderCode
+        #include <utility>
+        #include <vector>
+        using ItemGroup = PlacementStrategyLike<NfpPlacer>::ItemGroup; //Make an alias here since SIP doesn't namespace syntax nor template syntax.
+    %End
+
+    %ConvertFromTypeCode //From ItemGroup to Python list.
+        const long size = sipCpp->size();
+        if(size < 0)
+        {
+            return NULL;
+        }
+        PyObject* result_list = PyList_New(size);
+        if(!result_list)
+        {
+            return NULL;
+        }
+        for(size_t i = 0; i < size; ++i)
+        {
+            Item item = sipCpp->at(0);
+            PyObject* python_item = sipConvertFromInstance(&item, sipClass_Item, sipTransferObj);
+            if(!python_item)
+            {
+                Py_DECREF(result_list);
+                return NULL;
+            }
+            PyList_SET_ITEM(result_list, i, python_item);
+        }
+        return result_list;
+    %End
+
+    %ConvertToTypeCode //From Python list to ItemGroup.
+        PyObject* iterator = PyObject_GetIter(sipPy);
+        if(!sipIsErr) //We're only checking if it can be converted.
+        {
+            //Must be any iterable.
+            const bool iterable = (iterator != NULL);
+            Py_XDECREF(iterator);
+            return iterable;
+        }
+
+        ItemGroup* result = new ItemGroup();
+        PyObject* list_item;
+        while(list_item = PyIter_Next(iterator))
+        {
+            if(!sipCanConvertToInstance(list_item, sipClass_Item, SIP_NOT_NONE))
+            {
+                PyErr_Format(PyExc_TypeError, "Object in iterable cannot be converted to a libnest2d Item.");
+                *sipIsErr = 1;
+                break;
+            }
+            int state;
+            Item* item = reinterpret_cast<Item*>(sipConvertToInstance(list_item, sipClass_Item, 0, SIP_NOT_NONE, &state, sipIsErr));
+            if(!*sipIsErr)
+            {
+                result->push_back(*item);
```

Comment:
use `emplace_back(...)`

Reply:
`push_back` copies an existing item to the vector, while `emplace_back` constructs a new item from the constructor arguments, in-place in the vector's memory.
Because the pointer has a copy constructor, it is in this case effectively the same. But semantically I'm just adding an existing item to the vector rather than constructing a new one, so I think `push_back` is more appropriate.